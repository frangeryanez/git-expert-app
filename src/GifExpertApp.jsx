import { useState } from 'react';
import { AddCategory, GifGrid } from './components';

export const GifExpertApp = ({ defaultCategories=[] }) => {
  const [categories, setCategories] = useState(defaultCategories);

  const onAddCategory = newCategory => {
    if (categories.includes(newCategory.toLowerCase())) return;
    setCategories([newCategory, ...categories]);
  };

  return (
    <>
      <h1>Gif Expert App</h1>
      <hr />
      <AddCategory 
        // setCategories={ setCategories }
        onNewCategory={ value => onAddCategory(value) }
      />
      {categories?.map(category => (
        <GifGrid 
          category={ category } 
          key={ category } 
        />
      ))}
    </>
  );
};
