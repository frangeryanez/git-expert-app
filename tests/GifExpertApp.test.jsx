import { render } from '@testing-library/react';
import { GifExpertApp } from '../src/GifExpertApp';

describe('Test on file <GifExpertApp />', () => { 
  test('Must match the snapshot', () => {  
    const { container } = render(<GifExpertApp />);
    expect(container).toMatchSnapshot();
  });
});