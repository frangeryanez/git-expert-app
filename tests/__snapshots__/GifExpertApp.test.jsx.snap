// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test on file <GifExpertApp /> Must match the snapshot 1`] = `
<div>
  <h1>
    Gif Expert App
  </h1>
  <hr />
  <form
    aria-label="form"
  >
    <input
      type="text"
      value=""
    />
  </form>
</div>
`;
